src/main/host/managed_thread.rs

```diff
@@ -96,10 +100,11 @@ impl ManagedThread {
         match self.wait_for_next_event(ctx.host) {
             Ok(ShimEvent::SyscallComplete(res)) => res.retval,
             Err(SelfContainedChannelError::WriterIsClosed) => {
-                trace!("Plugin exited while executing native syscall {n}");
-                ctx.process.mark_as_exiting();
-                self.cleanup_after_exit_initiated();
-                SysCallReg::from(-(Errno::ESRCH as i64))
+                // This should only happen if the syscall was SYS_exit or
+                // SYS_exit_group, or if the process was killed externally, e.g.
+                // by sending a signal from outside of shadow. These cases would be
+                // tricky to support, and we shouldn't need to.
+                panic!("Plugin exited while executing native syscall {n}");
```

Comment:
Is it easy to get the exit status here and include it in the panic message? This might happen in practice if the oom manager kills some plugin process, which might be hard for the user to debug.